Sources/Event.swift

```diff
@@ -683,6 +711,88 @@ extension Signal.Event {
 			}
 		}
 	}
+
+	internal static func throttle(_ interval: TimeInterval, on scheduler: DateScheduler) -> Transformation<Value, Error> {
+		precondition(interval >= 0)
+
+		return { action in
+			let state: Atomic<ThrottleState<Value>> = Atomic(ThrottleState())
+			let schedulerDisposable = SerialDisposable()
```

Comment:
If this is used to create a `Signal` or `SignalProducer` and it is subsequently interrupted, it doesn't look like that interruption will propagate through to the scheduler here. (Unlike the previous implementation.)

Reply:
This is the current behaviour which is part of the operator lifting change (#140), although only `observe(on:)` is mentioned in the release note. The related issue was #227.

> interrupted now respects `observe(on:)`. (#140)
>
> When a produced Signal is interrupted, if observe(on:) is the last applied operator, interrupted would now be delivered on the Scheduler passed to observe(on:) just like other events.

Reply:
```
Disposals: ** <------------------ D1
                                  ** <--------------------------- D2
Events:    P1 -> map -> filter -> P2 -> map -> take -> observe -> P3
```

Disposing of D2 causes produced signal P2 to emit `interrupted` immediately (in general). The map-take-observe chain would then receive `interrupted`. P3 would not observe `interrupted` immediately, but the asynchrony and the serial event order are respected.

Reply:
Oh, I just realised I might have interpreted your question wrong.

> it doesn't look like that interruption will propagate through to the scheduler here.

We cannot interrupt an event transform specifically, so the interruption must be sourced from an upstream that can handle the interruption. In the implementation of `throttle`, all terminal events are scheduled to the given scheduler.

https://github.com/ReactiveCocoa/ReactiveSwift/pull/512/files/87592cd3f0968f35af8bd23f07cc02263ffad675#diff-c949eaeac664030204497a79420aaef3R724

Reply:
Thanks for the explanation. That helped a lot. 👍